src/allmydata/test/test_storage.py

```diff
@@ -540,21 +540,22 @@ def call_get_disk_stats(whichdir, reserved_space=0):
         OVERHEAD = 3*4
         LEASE_SIZE = 4+32+32+4
         canary = FakeCanary(True)
-        already,writers = self.allocate(ss, "vid1", [0,1,2], 1000, canary)
+        already, writers = self.allocate(ss, "vid1", [0,1,2], 1000, canary)
         self.failUnlessEqual(len(writers), 3)
         # now the StorageServer should have 3000 bytes provisionally
         # allocated, allowing only 2000 more to be claimed
         self.failUnlessEqual(len(ss._active_writers), 3)
 
         # allocating 1001-byte shares only leaves room for one
-        already2,writers2 = self.allocate(ss, "vid2", [0,1,2], 1001, canary)
+        already2, writers2 = self.allocate(ss, "vid2", [0,1,2], 1001, canary)
         self.failUnlessEqual(len(writers2), 1)
         self.failUnlessEqual(len(ss._active_writers), 4)
 
         # we abandon the first set, so their provisional allocation should be
         # returned
-        del already
-        del writers
```

Comment:
Okay, I misunderstood what the WeakKeyDictionary + test were doing. I've put them back, and added an explicit `gc.collect()` call to force PyPy to de-allocate those objects (on CPython they should go away immediately, because no more references). This works on my system, pushing this to see what CI says.